### PR TITLE
[release/8.0-staging] Fix calculation of channel bindings hash in managed NTLM implementation

### LINKED ISSUE #95725: UseManagedNtlm in NET8 cannot authenticate NTLM on IIS with Extended Protection Accept & Required
### Description

I have the same error issue with this issue [#19366](https://github.com/xamarin/xamarin-macios/issues/19366)
And I used UseManagedNtlm in the csproj file to solve it but I encountered another problem when using UseManagedNtlm on NET 8.

On ews on-premise with IIS, windows extended protection (WEP) have 3 level include Off, Accept, Required.
Extended_Protection

At NET 7, i have successfully authenticated with NTLM authentication when setting all 3 levels on IIS Server

After upgrading to NET8 with config  
```
<ItemGroup>
    <RuntimeHostConfigurationOption Include="System.Net.Security.UseManagedNtlm" Value="true" />
</ItemGroup>
```

I only successfully authenticated with NTLM authentication when setting level 1 (Off)
with level 2 & 3 (Accept, Required) when I perform NTLM authentication, I receive an error message that is "Unauthorized"
I want to be able to successfully authenticate at level 2 & 3 on IIS Server, what should I do?

### Reproduction Steps

Send request NTLM authenticate to server ews on-premise

### Expected behavior

none

### Actual behavior

none

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

Project sample:
[SocketRequestSample-Net8.zip](https://github.com/dotnet/runtime/files/13598496/SocketRequestSample-Net8.zip)

### PR DESCRIPTION
**Pull request by filipnavara**
Backport of #95898 to release/8.0-staging

Fixes #95725

## Customer Impact

Reported by customers in issue #95898 and by eM Client in https://github.com/dotnet/runtime/issues/95725#issuecomment-1875531103 - they cannot connect to Microsoft Exchange and quite a few other host providers on Android client (which uses ManagedNtlm by default).

Authenticating to HTTPS servers using NTLM fails with "Unauthorized" error code when Extended Authentication option is enabled on the server. This is an increasingly common setting on Microsoft Exchange deployments. The bug affects primarily the Android platform where the Managed NTLM implementation is used. We also allow the Managed NTLM as opt-in on macOS and Linux which would be likewise affected if the developer chooses to enable the `System.Net.Security.UseManagedNtlm` app context switch (which is rare).

## Regression

No

## Testing

Validated by eM Client on private Exchange instance in Azure VM + against one of the largest host providers in Germany.
The fix has been in .NET 9 since December 2023 and there were no observed issues.

Note: There are no unit tests, the set up is too complicated to test it meaningfully.

## Risk

Low; the modified code path is in platform specific code for a single authentication scheme.

### CODE REVIEW

**Comment by filipnavara** [comment]
/cc @rzikm @wfurt 

We have another customer of our product affected by this issue. They provided a test account on their server and we verified that the fix resolves the issue.

**Review by rzikm — APPROVED** [review]

**Comment by rzikm** [comment]
LGTM.

Removed Servicing-Consider for now, in case @karelz wants to edit the template first.

**Comment by JeroenBer** [comment]
Yes we urgently need this for our customers. Not only for Android but also iOS and macOS since the unmanaged implementations cause crashes and are not usable (#97966)

**Comment by JeroenBer** [comment]
@rzikm , not sure how this works but I see the PR is missing the "Servicing-approved" label before it can be merged. Who can do this ?

**Comment by JeroenBer** [comment]
@rzikm Thanks for keeping an eye on this. A lot of our customers are waiting for this. July is still a long time but please let's make that. If there's anything we can do to speed this up, like creating a service case at Microsoft, then please let me know.

**Comment by rzikm** [comment]
Creating a service case is unlikely to speed things up.

.NET has monthly cadence of servicing releases (with release dates coinciding with patch Tuesdays). Around the time one release is released, next one is branched off and starts to be prepared.

Restarting the process mid-way to include additional fixes would be costly (I don't know the figures, it's just my assumption based on my observations).

At any rate, there is potentially a way to unblock you and your customers earlier by using a private build of System.Net.Security.dll containing the fix. We use the process for validating that the fix is working. For desktop platforms the steps are simple (self-contained publish of the app+ overwriting the changed dll), I can provide you with the right dll for macOS if this option is available for you.

Unfortunately, I don't know if a similar process is possible for mobile platforms. Maybe @simonrozsival knows.

**Comment by simonrozsival** [comment]
> Unfortunately, I don't know if a similar process is possible for mobile platforms.

It is possible to use custom runtime build in a mobile app, but I wouldn't recommend it.

**Review by simonrozsival — APPROVED** [review]

**Comment by rzikm** [comment]
Approved via email on 06/03.

**Comment by JeroenBer** [comment]
@rzikm this is merged into dotnet:release/8.0-staging I think, does this mean it will come into the next servicing release in July ? 

For Android / iOS / macOS does this come available with a workload update for android / ios / macos or does this come available with a SDK update ? 
I mean Android / iOS / macOS have their own runtimes or is this something that is shared between all runtimes ? Or is this not in the runtime ?

**Comment by JeroenBer** [comment]
Thanks for the info, so just updating the SDK and rebuilding would be enough then ? No need to update workloads for android etc. For iOS / macOS we opted in for managed NTLM.

**Comment by wfurt** [comment]
> Thanks for the info, so just updating the SDK and rebuilding would be enough then ? No need to update workloads for android etc. For iOS / macOS we opted in for managed NTLM.

yes. cc @filipnavara just in case I missed something.

**Comment by filipnavara** [comment]
It’s correct. SDK update is all that’s needed.

**Pull request merged**
